### ISSUE TITLE
Cannot save a game

### ISSUE THREAD

**Issue by AMDmi3** [Andrettin/Wyrmgus#178]
With 5.2.1 I cannot save a game. Qt save dialog appears, I choose a name for a save game, then a message box appears saying "cannot save game to file: usr/home/marakasov/dvarwes.sav".
On the console:

```
Can't save to 'usr/home/marakasov/dwarves.sav'
```

It looks like it's generating an absolute path, but leading slash is removed for some reason, so it turns into broken relative path.

Also, the save dialog starts in my home directory, while I'd expect a dedicated directory for wyrmsun savegames created, e.g. `~/.local/share/wyrmsun/saves`, and the dialog suggesting to save games there.

**Comment by Andrettin** [comment]
The issue was possibly due to the way I converted from a "url" string to a filepath string in QML. I updated the places where I did that to use a better method, and hopefully this fixes the issue you experienced. Here are the relevant commits:

https://github.com/Andrettin/Wyrmsun/commit/eb938c40d432b05f238b729d88647be68e7096e1
https://github.com/Andrettin/Wyrmgus/commit/100cbc341f247010eed8fba7bd852933662cf818
https://github.com/Andrettin/Wyrmgus/commit/37a5f215d137e7a7bf3ae5e90fd3afa5a335e5b2

**Comment by Andrettin** [comment]
> 
> 
> With 5.2.2 saving works, thanks! However, the dialog still starts in the home dir, and it seems to be a Qt problem, as I've verified that the correct path is passed and also tried to hardcode specific directory in qml manually, nothing seems to work. Another problem is that it saves as `foo.gz`, when I print `foo` in the dialog, instead of `foo.sav.gz`

I'm happy that the saving problem got solved!

About the dialog path, which string did you pass to the "folder" property? I looked to see if anyone had a similar issue with the FileDialog, and I found this answer on Stack Overflow:

https://stackoverflow.com/questions/52931794/qml-set-filedialog-folder-property-to-valid-path-but-not-adhered-to

> Type of the FileDialog folder property is url. Now you are binding a local file path to that property. You can make a local file path a url by adding file:/// (Windows) or file:// (Linux/Mac) in front of the file path.

So perhaps writing it this way:

```
folder: "file://" + wyrmgus.save_path
```

Could solve the issue under Linux.

Or, alternatively, this could be a platform-independent way of writing it (it works on Windows at least):
```
folder: "file:" + wyrmgus.save_path
```

About the file extension, there's probably different behavior on Linux than on Windows. The Qt documentation says that the initial dot is removed:
https://doc.qt.io/qt-5/qml-qtquick-dialogs-filedialog.html#defaultSuffix-prop

But from what you described, it looks as if everything before the last dot is being removed. One way to solve this issue would be to use just ".gz" for the file extension on all platforms, as having two dots in a file name is error-prone, being liable to cause issues like the one you just mentioned.
